src/Mvc/Mvc.Core/src/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -441,8 +448,6 @@ public void Dispose()
                 _visitor.Metadata = _metadata;
                 _visitor.Model = _model;
                 _visitor.Strategy = _strategy;
-
-                _visitor._currentPath.Pop(_newModel);
```

Comment:
This part seems like a big red flag (in the existing code, not in your changes). Seeing pushes and pops so disconnected from each other is a good clue that we had a bug here, your changes make this much easier to reason about 👍